### ISSUE TITLE
eval_mcc is 0.0 for CoLA at every epoch

### ISSUE THREAD

**Issue by sandy9999** [Hazelsuko07/TextHide#2]
I made a couple of commits in my fork to use GLUE data hosted at a different place, and to force the use of PyTorch even when Tensorflow is installed. <https://github.com/sandy9999/TextHide/tree/workaround_glue_data>

Steps to reproduce (I'm using a Google Colab instance):
- ```!git clone -b workaround_glue_data https://github.com/sandy9999/TextHide```
- ```%cd TextHide```
- ```!python download_glue_data.py --data_dir glue_data --tasks all```
- ```!pip install -r requirements.txt```
- ```sh
    #!export GLUE_DIR=glue_data
    #!export TASK_NAME=SST-2
    %env GLUE_DIR=glue_data
    %env TASK_NAME=CoLA
    !python run_glue.py \
    --model_name_or_path bert-base-cased \
    --task_name $TASK_NAME \
    --do_train \
    --do_eval \
    --data_dir $GLUE_DIR/$TASK_NAME \
    --max_seq_length 128 \
    --per_device_train_batch_size 32 \
    --learning_rate 1e-5 \
    --dropout 0.4 \
    --num_train_epochs 20.0 \
    --num_k 4 \
    --num_sigma 256 \
    --output_dir ./results/$TASK_NAME/BERT_256_4_intra/ \
    --overwrite_output_dir
    ```
![image](https://user-images.githubusercontent.com/17477947/107004950-0aa10b80-67b5-11eb-8f9b-a7b70c211296.png)

**Comment by Hazelsuko07** [comment]
Hi Sandy,

Thanks for raising the issue. 

We used different hyperparams for different GLUE datasets / learning settings. Sorry that I didn't clarify this in the main repo. 

For CoLA with the BERT base cased model (intra-dataset, m=256, k=4), maybe you can try this set of hyperparam: 

> --max_seq_length 128   --per_device_train_batch_size 32   --learning_rate 1e-5   --dropout 0   --num_train_epochs 30.0

This should give you an eval_mcc of around 60%.

Please let me know if you still have trouble achieving a similar result. Thanks!

**Comment by sandy9999** [comment]
Yes, I'm able to achieve this result for CoLA. Thank you very much! It would be helpful if we could have the set of hyper params corresponding to the expected accuracy/mcc for every dataset.

**Comment by Hazelsuko07** [comment]
Thanks for the suggestion! I will update the repo :)

**Comment by ishanisri** [comment]
Hi, continuing on this thread, I have been trying to run STS-B and RTE but the results are drastically different from what is mentioned in the paper. Can you tell the hyperparameters that were used for the same?

**Comment by Hazelsuko07** [comment]
> Hi, continuing on this thread, I have been trying to run STS-B and RTE but the results are drastically different from what is mentioned in the paper. Can you tell the hyperparameters that were used for the same?

Hi, 

Thanks for reminding me! I just pushed the bash scripts for all datasets: https://github.com/Hazelsuko07/TextHide/commit/e733e2fab84942a0038b2ee66095f898de1b3f12. Please let me know if there are further questions 😃 

Best,
Yangsibo
